optimum/exporters/tflite/convert.py

```diff
@@ -0,0 +1,183 @@
+            continue
+
+        ref_output = ref_outputs[name].numpy()
+
+        logger.info(f'\t- Validating TFLite Model output "{name}":')
+
+        # Shape
+        if not output.shape == ref_output.shape:
+            logger.error(f"\t\t-[x] shape {output.shape} doesn't match {ref_output.shape}")
+            shape_failures.append((name, ref_output.shape, output.shape))
+        else:
+            logger.info(f"\t\t-[✓] {output.shape} matches {ref_output.shape}")
+
+        # Values
+        if not np.allclose(ref_output, output, atol=atol):
+            max_diff = np.amax(np.abs(ref_output - output))
+            logger.error(f"\t\t-[x] values not close enough, max diff: {max_diff} (atol: {atol})")
+            value_failures.append((name, max_diff))
+        else:
+            logger.info(f"\t\t-[✓] all values close (atol: {atol})")
+
+    if shape_failures:
+        msg = "\n".join(f"- {t[0]}: got {t[1]} (reference) and {t[2]} (TFLite)" for t in shape_failures)
+        raise ShapeError(
+            f"Output shapes do not match between reference model and TensorFlow Lite exported model:\n" "{msg}"
+        )
+
+    if value_failures:
+        msg = "\n".join(f"- {t[0]}: max diff = {t[1]}" for t in value_failures)
+        raise AtolError(
+            "The maximum absolute difference between the output of the reference model and the TensorFlow Lite "
+            f"exported model is not within the set tolerance {atol}:\n{msg}"
+        )
+
+
+def export(
+    model: "TFPreTrainedModel",
+    config: "TFLiteConfig",
+    output: Path,
+) -> Tuple[List[str], List[str]]:
+    """
+    Exports a TensorFlow model to a TensorFlow Lite model.
+
+    Args:
+        model ([`TFPreTrainedModel`]):
+            The model to export.
+        config ([`~exporters.tflite.TFLiteConfig`]):
+            The TFLite configuration associated with the exported model.
+        output (`Path`):
+            Directory to store the exported TFLite model.
+
+    Returns:
+        `Tuple[List[str], List[str]]`: A tuple with an ordered list of the model's inputs, and the named inputs from
+        the TFLite configuration.
+    """
+    if not is_tf_available():
+        raise ImportError("Cannot convert because TensorFlow is not installed. " "Please install TensorFlow first.")
+    import tensorflow as tf
+
+    output.parent.mkdir(parents=True, exist_ok=True)
+
+    logger.info(f"Using TensorFlow: {tf.__version__}")
+    model.config.return_dict = True
+
+    # Check if we need to override certain configuration item
+    if config.values_override is not None:
+        logger.info(f"Overriding {len(config.values_override)} configuration item(s)")
+        for override_config_key, override_config_value in config.values_override.items():
+            logger.info(f"\t- {override_config_key} -> {override_config_value}")
+            setattr(model.config, override_config_key, override_config_value)
+
+    signatures = config.model_to_signatures(model)
```

Comment:
We should be able to call `save()` directly on the model to save it as a `SavedModel` resource unless there's an important reason not to. Ccing @Rocketknight1 @gante. 

Also, do we need to run the intermediate save? I can imagine it's likely there for better standardization of the modules but saving a TF model into SavedModel takes a lot of time (because of the compilation involved). 

Could we directly take a `TFPretrainedModel` or even its concrete function and use that for the TFLite export ([here's](https://colab.research.google.com/gist/sayakpaul/6e4f06232b689afa74a43b1b8e2be243/scratchpad.ipynb) a simple example of obtaining concrete function from a Keras model directly)?

Reply:
> I could not find a way to provide signatures in that case

Which case? Could you expand a bit more? 

> It should not provide a speedup in the end because it is (or it more precisely it will be) mandatory to provide a trackable object when exporting with a concrete function, which in turn will do the export to a saved model in the backend.

Could you elaborate a bit more on this? Why providing the trackable object is a requirement?

Reply:
- I could not find a way to provide signatures when exporting from a concrete function.
- When exporting from a concrete function, if you dont provide a trackable object, it kind of works, but a warning tells you that it will be deprecated => you can also see this warning from your [code sample](https://colab.research.google.com/gist/sayakpaul/6e4f06232b689afa74a43b1b8e2be243/scratchpad.ipynb#scrollTo=KLuOmvf5Fm0B&line=2&uniqifier=1)

Reply:
Yes, that's not an issue, you can specify the signature you want, that's what I did.

Reply:
Ah, sorry, I was looking at an old version of the page and didn't see the up-to-date conversation!

To support signatures when exporting from a concrete function, you need to make one concrete function per signature. That's exactly what a `ConcreteFunction` is in TensorFlow - it's a function with one precisely defined signature.

Also, if you're getting errors about a missing `trackable_obj`, I'm 99% sure the `trackable_obj` you need is just the original `model` object. [Here](https://colab.research.google.com/gist/Rocketknight1/781023e44fd88f62b67d3c46a7cd50f9/scratchpad.ipynb) is an updated code sample that uses multiple concrete functions and doesn't have the error.

Reply:
Ah, sorry, I follow now! Yeah, I don't know of any good way to name the signatures when using concrete functions.

Reply:
For reference in `optimum.tir` I was generating on the fly concrete definition with custom signatures, maybe it can help (please dont judge me ...):

```python
    def export_model_to_mlir(self, model: TFPreTrainedModel, examples: Optional = None, dynamic_axes: List[int] = None):
        # Define function signature
        if examples is not None:
            serving_signature = {
                "input_ids": tf.TensorSpec((1, examples[0].shape[1]), dtype=tf.dtypes.int32, name="input_ids"),
                "attention_mask": tf.TensorSpec((1, examples[1].shape[1]), dtype=tf.dtypes.int8, name="attention_mask"),
                # "token_type_ids": tf.TensorSpec((1, None), dtype=tf.dtypes.int8, name="token_type_ids")
            }
        else:
            serving_signature = {
                "input_ids": tf.TensorSpec((1, None), dtype=tf.dtypes.int32, name="input_ids"),
                "attention_mask": tf.TensorSpec((1, None), dtype=tf.dtypes.int8, name="attention_mask"),
                # "token_type_ids": tf.TensorSpec((1, None), dtype=tf.dtypes.int8, name="token_type_ids")
            }

        # Traced function to inject at module-level
        @tf.function(input_signature=[serving_signature])
        def forward(self, inputs):
            outputs = self.eager_serving(inputs)
            return model.serving_output(outputs)

        # Bind the function on the module (TODO: my eyes are already crying blood at this stage ...)
        model.forward = forward.__get__(model, type(model))
```